### PR TITLE
Feature/ioc boot

### PR DESCRIPTION
**Pull request by BenBradnick**
This PR contains a set of small changes based on generating the ioc boot script:

1. Update docs to reflect ibek build-helm no longer needs a schema arg
2. Added 2 elements to the IOC boot script - environment variables and pos iocInit entries. An epics.ibek.yaml file was created with entities which have entries for these new elements, as well as a test for generating a boot script containing them
3. Tidied up the existing ioc boot Jinja template to remove duplicates and improve spacing
4. Fixed test in test_cli for building IOC against multiple support module definition files

### CODE REVIEW

File: tests/samples/yaml/epics.ibek.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# yaml-language-server: $schema=file://schemas/ibek.schema.json
+module: epics
+definitions:
+  - name: EPICS_CA_MAX_ARRAY_BYTES
```
Comment (gilesknap) [inline tests/samples/yaml/epics.ibek.yaml:4]: We have the ability to add arbitrary dbpf's but it seems not arbitrary env_vars. Would this be a good place to put them?

File: tests/samples/yaml/epics.ibek.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# yaml-language-server: $schema=file://schemas/ibek.schema.json
```
Comment (gilesknap) [inline tests/samples/yaml/epics.ibek.yaml:1]: I like epics.ibek.yaml.

There are probably other things that would go in here I'm guessing.

**Review by gilesknap — APPROVED** [review]
This looks great.

File: tests/samples/yaml/epics.ibek.yaml (hunk 1 above)
Comment (gilesknap) [reply tests/samples/yaml/epics.ibek.yaml:4]: If so this should be in addtion to EPICS_CA_MAX_ARRAY_BYTES and any other variables that we expect the majority of IOCs to set.

Comment (BenBradnick) [reply tests/samples/yaml/epics.ibek.yaml:4]: Sure, I think this is a good place

File: tests/samples/yaml/epics.ibek.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+module: epics
+definitions:
+  - name: EPICS_CA_MAX_ARRAY_BYTES
+    args:
+      - type: int
+        name: max_bytes
+        description: Max size in bytes for sending arrays over channel access
+        default: 6000000
+    env_vars:
+      - epicsEnvSet "EPICS_CA_MAX_ARRAY_BYTES", '{{ max_bytes }}'
```
Comment (thomascobb) [inline tests/samples/yaml/epics.ibek.yaml:11]: I'd prefer if we didn't have `epicsEnvSet` in there, how about we change from `List[str]` to `Dict[str, str]`:
```suggestion
      EPICS_CA_MAX_ARRAY_BYTES: {{ max_bytes }}
```
Although then we should also change `define_args` to a `Dict[str, str]` to match

File: ibek/templates/ioc.boot.jinja [hunk 4]
```diff
@@ -1,12 +1,13 @@
 cd "$(TOP)"
-epicsEnvSet "EPICS_CA_MAX_ARRAY_BYTES", '6000000'
-epicsEnvSet "EPICS_TS_MIN_WEST", '0'
-cd "$(TOP)"
+{% if env_var_elements %}
+{{ env_var_elements -}}
```
[the remark this answers is not in the document]
Comment (BenBradnick) [reply ibek/templates/ioc.boot.jinja:3]: It's syntax for control structures, in this case I just check to see if env_var_elements is defined - https://jinja.palletsprojects.com/en/3.0.x/templates/#list-of-control-structures

This was just to keep the boot script tidy, as I found it difficult to control new lines with optional elements.

File: tests/samples/yaml/epics.ibek.yaml (hunk 3 above)
Comment (BenBradnick) [reply tests/samples/yaml/epics.ibek.yaml:11]: That does look tidier, I'll take a look

File: tests/samples/yaml/epics.ibek.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      - type: str
+        name: pv
+        description: Name of PV
+      - type: str
+        name: value
+        description: Value to set
+    post_ioc_init:
+      - dbpf "{{ pv }}", "{{ value }}"
+
+  - name: epicsEnvSet
```
Comment (BenBradnick) [inline tests/samples/yaml/epics.ibek.yaml:35]: @thomascobb is this better for defining environment variables? Entities for specific variables then have the name hard-coded. One limitation is that all values are interpreted as strings.

Comment (thomascobb) [reply tests/samples/yaml/epics.ibek.yaml:35]: Yes, I think this is fine

**Pull request merged**